test/test-parse-function.cpp

```diff
@@ -233,6 +233,35 @@ TEST(test_parse, generator_function_statement) {
   }
 }
 
+TEST(test_parse, unused_arrow_expression) {
+  {
+    padded_string code(u8"async () => {}"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    p.parse_and_visit_module(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",       //
+                                      "visit_enter_function_scope_body",  //
+                                      "visit_exit_function_scope",        //
+                                      "visit_end_of_module"));
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_OFFSETS(
+                              &code, error_unused_arrow_expression,  //
+                              where, strlen(u8""), u8"async")));
+  }
+  {
+    padded_string code(u8"() => {}"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    p.parse_and_visit_module(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",       //
+                                      "visit_enter_function_scope_body",  //
+                                      "visit_exit_function_scope",        //
+                                      "visit_end_of_module"));
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_OFFSETS(
+                              &code, error_unused_arrow_expression,  //
+                              where, strlen(u8""), u8"(")));
+  }
+}
```

Comment:
I think the following code should also report warning E0211. The programmer probably meant to write the IIFE pattern but forgot to call the function:
```js
(() => {
  console.log('hi');
});
```